slatedb/src/error.rs

```diff
@@ -483,6 +506,9 @@ impl From<SlateDBError> for Error {
             SlateDBError::BackgroundTaskExists(_) => Error::internal(msg),
             SlateDBError::BackgroundTaskCancelled(_) => Error::internal(msg),
             SlateDBError::BackgroundTaskExecutorStarted => Error::internal(msg),
+            SlateDBError::TransactionalObjectError(err) => {
```

Comment:
I think this should be a data error. The comment above says:

> // returned when the persisted state is invalid (e.g. malformed name, missing file)

Which fits exactly with our data error definition.

Reply:
oh yeah that's a bug. Fixing.